### ISSUE TITLE
Trying to install coreutils breaks container

### ISSUE THREAD

**Issue by carylewis** [archlinux/archlinux-docker#28]
Running a archlinux container:

`docker run -i -t archlinux /bin/bash`

Then running:

`pacman -Sy`

Then running: 

```
pacman -S coreutils
warning: coreutils-8.31-3 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) coreutils-8.31-3

Total Download Size:    2.37 MiB
Total Installed Size:  15.39 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 coreutils-8.31-3...     2.4 MiB  4.06 MiB/s 00:01 [######################] 100%
(1/1) checking keys in keyring                     [######################] 100%
(1/1) checking package integrity                   [######################] 100%
(1/1) loading package files                        [######################] 100%
(1/1) checking for file conflicts                  [######################] 100%
:: Processing package changes...
error: could not remove database entry coreutils-8.31-3
error: could not create directory /var/lib/pacman/local/coreutils-8.31-3/: File exists
error: could not commit transaction
error: failed to commit transaction (transaction aborted)
Errors occurred, no packages were upgraded.

```
Results in the basic file commands not working?

`[root@11baedf72d8d /]# ls
bash: ls: command not found
[root@11baedf72d8d /]# 
`

How do you install utilities such as diff, etc. in an archlinux container?

**Comment by SantiagoTorres** [comment]
Hmm, this is indeed strange. You should be able to run this and (as you saw) coreutils is already installed. Diffutils should give you diff, yet I wonde why this bug happened to begin with. Thank you for reporting!

**Comment by SantiagoTorres** [comment]
Hi @carylewis. Does this happen if you use `pacman -Syu` instead of `pacman -Sy`? Arch [doesn't support partial upgrades](https://wiki.archlinux.org/index.php/System_maintenance#Partial_upgrades_are_unsupported), and I wonder if that's part of the issue

**Comment by carylewis** [comment]
same error:

pacman -Syu
:: Synchronizing package databases...
 core                  133.9 KiB   740 KiB/s 00:00 [######################]
100%
 extra                1641.8 KiB  2.68 MiB/s 00:01 [######################]
100%
 community               4.8 MiB  8.27 MiB/s 00:01 [######################]
100%
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (7) gnutls-3.6.12-1  iana-etc-20200117-1  libtasn1-4.16.0-1
             pacman-mirrorlist-20200207-1  shadow-4.8.1-1  systemd-244.2-1
             systemd-libs-244.2-1

Total Download Size:    9.97 MiB
Total Installed Size:  37.25 MiB
Net Upgrade Size:       0.11 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 iana-etc-2020011...   387.1 KiB  1166 KiB/s 00:00 [######################]
100%
 libtasn1-4.16.0-...   121.3 KiB   809 KiB/s 00:00 [######################]
100%
 systemd-libs-244...   457.1 KiB  1539 KiB/s 00:00 [######################]
100%
 shadow-4.8.1-1-x...  1171.2 KiB  2.64 MiB/s 00:00 [######################]
100%
 systemd-244.2-1-...     5.1 MiB  8.90 MiB/s 00:01 [######################]
100%
 gnutls-3.6.12-1-...     2.8 MiB  6.38 MiB/s 00:00 [######################]
100%
 pacman-mirrorlis...     6.0 KiB  0.00   B/s 00:00 [######################]
100%
(7/7) checking keys in keyring                     [######################]
100%
(7/7) checking package integrity                   [######################]
100%
(7/7) loading package files                        [######################]
100%
(7/7) checking for file conflicts                  [######################]
100%
:: Processing package changes...
error: could not remove database entry iana-etc-20191231-1
(1/7) upgrading iana-etc                           [######################]
100%
error: could not remove database entry libtasn1-4.15.0-1
(2/7) upgrading libtasn1                           [######################]
100%
error: could not remove database entry systemd-libs-244.1-1
(3/7) upgrading systemd-libs                       [######################]
100%
error: could not remove database entry shadow-4.8-1
(4/7) upgrading shadow                             [######################]
100%
error: could not remove database entry systemd-244.1-1
(5/7) upgrading systemd                            [######################]
100%
error: could not remove database entry gnutls-3.6.11.1-1
(6/7) upgrading gnutls                             [######################]
100%
error: could not remove database entry pacman-mirrorlist-20200103-1
(7/7) upgrading pacman-mirrorlist                  [######################]
100%
warning: /etc/pacman.d/mirrorlist installed as
/etc/pacman.d/mirrorlist.pacnew
ldconfig: Cannot lstat /usr/lib/libgnutls.so.30.26.1: No such file or
directory
ldconfig: Cannot lstat /usr/lib/libtasn1.so.6.5.7: No such file or directory
:: Running post-transaction hooks...
(1/9) Creating system user accounts...
(2/9) Updating journal message catalog...
(3/9) Reloading system manager configuration...
  Skipped: Current root is not booted.
(4/9) Updating udev hardware database...
(5/9) Applying kernel sysctl settings...
  Skipped: Current root is not booted.
(6/9) Creating temporary files...
[/usr/lib/tmpfiles.d/journal-nocow.conf:26] Failed to resolve specifier:
uninitialized /etc detected, skipping
All rules containing unresolvable specifiers will be skipped.
(7/9) Reloading device manager configuration...
  Skipped: Device manager is not running.
(8/9) Arming ConditionNeedsUpdate...
(9/9) Reloading system bus configuration...
  Skipped: Current root is not booted.

[root@b13750d79918 /]# pacman -S coreutils
error: duplicated database entry 'iana-etc'
error: duplicated database entry 'libtasn1'
error: duplicated database entry 'systemd-libs'
error: duplicated database entry 'shadow'
error: duplicated database entry 'systemd'
error: duplicated database entry 'gnutls'
error: duplicated database entry 'pacman-mirrorlist'
warning: coreutils-8.31-3 is up to date -- reinstalling
resolving dependencies...
error: could not open file /var/lib/pacman/local/gnutls-3.6.11.1-1/desc: No
such file or directory
error: could not open file /var/lib/pacman/local/iana-etc-20191231-1/desc:
No such file or directory
error: could not open file /var/lib/pacman/local/libtasn1-4.15.0-1/desc: No
such file or directory
looking for conflicting packages...
error: could not open file
/var/lib/pacman/local/pacman-mirrorlist-20200103-1/desc: No such file or
directory
error: could not open file /var/lib/pacman/local/shadow-4.8-1/desc: No such
file or directory
error: could not open file /var/lib/pacman/local/systemd-244.1-1/desc: No
such file or directory
error: could not open file /var/lib/pacman/local/systemd-libs-244.1-1/desc:
No such file or directory

Packages (1) coreutils-8.31-3

Total Download Size:    2.37 MiB
Total Installed Size:  15.39 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 coreutils-8.31-3...     2.4 MiB  4.25 MiB/s 00:01 [######################]
100%
(1/1) checking keys in keyring                     [######################]
100%
(1/1) checking package integrity                   [######################]
100%
(1/1) loading package files                        [######################]
100%
(1/1) checking for file conflicts                  [######################]
100%
:: Processing package changes...
error: could not remove database entry coreutils-8.31-3
error: could not create directory /var/lib/pacman/local/coreutils-8.31-3/:
File exists
error: could not commit transaction
error: failed to commit transaction (transaction aborted)
Errors occurred, no packages were upgraded.
[root@b13750d79918 /]# ls
bash: ls: command not found

Installing diffutils worked

I'm not sure if this error is part of the problem or note:

warning: could not fully load metadata for package gnutls-3.6.11.1-1

**Comment by SantiagoTorres** [comment]
huh, that's strange. Can you tell me what the image id it is that you're running?:

```
docker image ls archlinux
```

should probably do it.

**Comment by carylewis** [comment]
```
docker images --format "{{.ID}}: {{.Repository}} {{.Tag}}  {{.Digest}}"  archlinux
8635d7bd071c: archlinux latest  sha256:01333c9d0acd59c4566d7e4138384d992cb16079530d4198d33916e4397e0230
```

**Comment by SantiagoTorres** [comment]
Hmmm, this is the right digest.

 I'm quite confused as I can't reproduce on my side. Was this on a new container or did you use the one with the partial update? I was able to install all of `gnutls` `coreutils` and `diffutils` without any issues on three different machines already...

**Comment by carylewis** [comment]
On my mac:

`docker --version
Docker version 19.03.5, build 633a0ea`

This is command I am running (I am running docker on a mac):
`docker run -it archlinux:latest bash
Unable to find image 'archlinux:latest' locally
latest: Pulling from library/archlinux
ef42853c0f87: Pull complete 
e5cd2121e98d: Pull complete 
214eaa4433b2: Pull complete 
56470d7667b7: Pull complete 
896334b8e583: Pull complete 
Digest: sha256:01333c9d0acd59c4566d7e4138384d992cb16079530d4198d33916e4397e0230
Status: Downloaded newer image for archlinux:latest
[root@ebe167e74689 /]# `

`pacman -Syu`
works, and 
`pacman -S coreutils`
works

But after removing all docker containers on my centos ec2 machine:
`docker run -it archlinux:latest bash
Unable to find image 'archlinux:latest' locally
latest: Pulling from library/archlinux
ef42853c0f87: Pull complete 
e5cd2121e98d: Pull complete 
214eaa4433b2: Pull complete 
56470d7667b7: Pull complete 
896334b8e583: Pull complete 
Digest: sha256:01333c9d0acd59c4566d7e4138384d992cb16079530d4198d33916e4397e0230
Status: Downloaded newer image for archlinux:latest
[root@e02ed1b1fd9d /]# 
[root@e02ed1b1fd9d /]# 
[root@e02ed1b1fd9d /]# ls
bin   dev  home  lib64	opt   root  sbin  sys  usr
boot  etc  lib	 mnt	proc  run   srv   tmp  var
[root@e02ed1b1fd9d /]# pacman -Syu
:: Synchronizing package databases...
 core                  134.1 KiB   888 KiB/s 00:00 [######################] 100%
 extra                1641.6 KiB  4.06 MiB/s 00:00 [######################] 100%
 community               4.8 MiB  7.52 MiB/s 00:01 [######################] 100%
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (8) ca-certificates-mozilla-3.50-1  gnutls-3.6.12-1
             iana-etc-20200117-1  libtasn1-4.16.0-1
             pacman-mirrorlist-20200207-1  shadow-4.8.1-1  systemd-244.2-2
             systemd-libs-244.2-2

Total Download Size:   10.30 MiB
Total Installed Size:  38.17 MiB
Net Upgrade Size:       0.11 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 iana-etc-2020011...   387.1 KiB  1170 KiB/s 00:00 [######################] 100%
 libtasn1-4.16.0-...   121.3 KiB   926 KiB/s 00:00 [######################] 100%
 systemd-libs-244...   457.1 KiB  1329 KiB/s 00:00 [######################] 100%
 shadow-4.8.1-1-x...  1171.2 KiB  2.36 MiB/s 00:00 [######################] 100%
 systemd-244.2-2-...     5.1 MiB  8.35 MiB/s 00:01 [######################] 100%
 ca-certificates-...   336.7 KiB  1285 KiB/s 00:00 [######################] 100%
 gnutls-3.6.12-1-...     2.8 MiB  3.98 MiB/s 00:01 [######################] 100%
 pacman-mirrorlis...     6.0 KiB  0.00   B/s 00:00 [######################] 100%
(8/8) checking keys in keyring                     [######################] 100%
(8/8) checking package integrity                   [######################] 100%
(8/8) loading package files                        [######################] 100%
(8/8) checking for file conflicts                  [######################] 100%
:: Processing package changes...
error: could not remove database entry iana-etc-20191231-1
(1/8) upgrading iana-etc                           [######################] 100%
error: could not remove database entry libtasn1-4.15.0-1
(2/8) upgrading libtasn1                           [######################] 100%
error: could not remove database entry systemd-libs-244.1-1
(3/8) upgrading systemd-libs                       [######################] 100%
error: could not remove database entry shadow-4.8-1
(4/8) upgrading shadow                             [######################] 100%
error: could not remove database entry systemd-244.1-1
(5/8) upgrading systemd                            [######################] 100%
error: could not remove database entry ca-certificates-mozilla-3.49.2-3
(6/8) upgrading ca-certificates-mozilla            [######################] 100%
error: could not remove database entry gnutls-3.6.11.1-1
(7/8) upgrading gnutls                             [######################] 100%
error: could not remove database entry pacman-mirrorlist-20200103-1
(8/8) upgrading pacman-mirrorlist                  [######################] 100%
warning: /etc/pacman.d/mirrorlist installed as /etc/pacman.d/mirrorlist.pacnew
ldconfig: Cannot lstat /usr/lib/libgnutls.so.30.26.1: No such file or directory
ldconfig: Cannot lstat /usr/lib/libtasn1.so.6.5.7: No such file or directory
:: Running post-transaction hooks...
( 1/10) Creating system user accounts...
( 2/10) Updating journal message catalog...
( 3/10) Reloading system manager configuration...
  Skipped: Current root is not booted.
( 4/10) Updating udev hardware database...
( 5/10) Applying kernel sysctl settings...
  Skipped: Current root is not booted.
( 6/10) Creating temporary files...
[/usr/lib/tmpfiles.d/journal-nocow.conf:26] Failed to resolve specifier: uninitialized /etc detected, skipping
All rules containing unresolvable specifiers will be skipped.
( 7/10) Reloading device manager configuration...
  Skipped: Device manager is not running.
( 8/10) Arming ConditionNeedsUpdate...
( 9/10) Reloading system bus configuration...
  Skipped: Current root is not booted.
(10/10) Rebuilding certificate stores...`
and then trying to run:
`pacman -S coreutils
error: duplicated database entry 'iana-etc'
error: duplicated database entry 'libtasn1'
error: duplicated database entry 'systemd-libs'
error: duplicated database entry 'shadow'
error: duplicated database entry 'systemd'
error: duplicated database entry 'ca-certificates-mozilla'
error: duplicated database entry 'gnutls'
error: duplicated database entry 'pacman-mirrorlist'
warning: coreutils-8.31-3 is up to date -- reinstalling
resolving dependencies...
error: could not open file /var/lib/pacman/local/ca-certificates-mozilla-3.49.2-3/desc: No such file or directory
error: could not open file /var/lib/pacman/local/gnutls-3.6.11.1-1/desc: No such file or directory
error: could not open file /var/lib/pacman/local/iana-etc-20191231-1/desc: No such file or directory
error: could not open file /var/lib/pacman/local/libtasn1-4.15.0-1/desc: No such file or directory
looking for conflicting packages...
error: could not open file /var/lib/pacman/local/pacman-mirrorlist-20200103-1/desc: No such file or directory
error: could not open file /var/lib/pacman/local/shadow-4.8-1/desc: No such file or directory
error: could not open file /var/lib/pacman/local/systemd-244.1-1/desc: No such file or directory
error: could not open file /var/lib/pacman/local/systemd-libs-244.1-1/desc: No such file or directory

Packages (1) coreutils-8.31-3

Total Download Size:    2.37 MiB
Total Installed Size:  15.39 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 coreutils-8.31-3...     2.4 MiB  4.44 MiB/s 00:01 [######################] 100%
(1/1) checking keys in keyring                     [######################] 100%
(1/1) checking package integrity                   [######################] 100%
(1/1) loading package files                        [######################] 100%
(1/1) checking for file conflicts                  [######################] 100%
:: Processing package changes...
error: could not remove database entry coreutils-8.31-3
error: could not create directory /var/lib/pacman/local/coreutils-8.31-3/: File exists
error: could not commit transaction
error: failed to commit transaction (transaction aborted)
Errors occurred, no packages were upgraded.
[root@e02ed1b1fd9d /]# ls
bash: /usr/sbin/ls: No such file or directory`

I can't explain why docker is behaving differently on ec2 vm vs. on a mac, perhaps the kernel version?

On ec2:
`uname -a
Linux 7a92f67db8cd 3.10.0-1062.12.1.el7.x86_64 #1 SMP Tue Feb 4 23:02:59 UTC 2020 x86_64 GNU/Linux
`

**Comment by SantiagoTorres** [comment]
Strange, sorry for asking all of this but care to share the output of `docker info` on both boxes?

**Comment by carylewis** [comment]
On the mac:
```
docker info
Client:
 Debug Mode: false

Server:
 Containers: 21
  Running: 0
  Paused: 0
  Stopped: 21
 Images: 31
 Server Version: 19.03.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.943GiB
 Name: docker-desktop
 ID: OV64:FI2M:TQLR:CFNT:WNIT:5FX2:KN55:SNRO:527H:VN4J:3UWG:DIRJ
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 34
  Goroutines: 51
  System Time: 2020-02-09T16:52:34.5619384Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

On ec2:

```
docker info
Client:
 Debug Mode: false

Server:
 Containers: 5
  Running: 0
  Paused: 0
  Stopped: 5
 Images: 3
 Server Version: 19.03.5
 Storage Driver: overlay
  Backing Filesystem: xfs
  Supports d_type: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1062.12.1.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 464.9MiB
 Name: ip-172-31-74-101.ec2.internal
 ID: IDFU:DA6F:DB57:NTF2:E27H:GHA5:TLEB:CORU:GG3Y:CXZF:5ELI:PLQK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: the overlay storage-driver is deprecated, and will be removed in a future release.
WARNING: overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior.
         Reformat the filesystem with ftype=1 to enable d_type support.
         Running without d_type support will not be supported in future releases.
```
Seeing the warnings, I uninstalled docker-ce and reinstalled, and now the arch linux container works the same as on the Mac.

Sorry for all the noise, but maybe a warning on the main page telling people to check their docker installation via docker info to make sure that docker will work correctly.

Here is the the docker info after the de/re install:

```
docker info
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 19.03.5
 Storage Driver: devicemapper
  Pool Name: docker-259:1-92284641-pool
  Pool Blocksize: 65.54kB
  Base Device Size: 10.74GB
  Backing Filesystem: xfs
  Udev Sync Supported: true
  Data file: /dev/loop0
  Metadata file: /dev/loop1
  Data loop file: /var/lib/docker/devicemapper/devicemapper/data
  Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
  Data Space Used: 578.1MB
  Data Space Total: 107.4GB
  Data Space Available: 4.944GB
  Metadata Space Used: 17.96MB
  Metadata Space Total: 2.147GB
  Metadata Space Available: 2.13GB
  Thin Pool Minimum Free Space: 10.74GB
  Deferred Removal Enabled: true
  Deferred Deletion Enabled: true
  Deferred Deleted Device Count: 0
  Library Version: 1.02.158-RHEL7 (2019-05-13)
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1062.12.1.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 464.9MiB
 Name: ip-172-31-74-101.ec2.internal
 ID: IDFU:DA6F:DB57:NTF2:E27H:GHA5:TLEB:CORU:GG3Y:CXZF:5ELI:PLQK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: the devicemapper storage-driver is deprecated, and will be removed in a future release.
WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
```

**Comment by SantiagoTorres** [comment]
Thanks for figuring this out, @carylewis! at least now we know what to check for if we run into these issues from now on :)
